### PR TITLE
Add stacked octree observations

### PR DESCRIPTION
**Pull request by AndrejOrsula**
This PR adds option to stack multiple consequent octree observations, similar to Atari stacked frames. This hopefully gives agent a sense of a temporal information from the scene from which it might be able to deduct dynamics.

For the first n observations after reset, octrees are repeated until the buffer is filled.

It is optimised to be processed as a single batch. Therefore, there is little performance loss on system with GPU that has excess of unused pipelines, even if this feature is enabled. 

However, memory usage is effectively multiplied. This is something that would be nice to optimise (the complexity is however out of the scope for this project, as it would require rewriting large portion of off-policy algorithm handling)

### CODE REVIEW

**Comment by AndrejOrsula** [comment]
**Important:** I have just realised that this approach will NOT function properly with network architectures that use batch normalisation (e.g. if `OctreeCnnPolicy`'s `batch_normalization=True`), as it normalizes the stacked frames.

Running some tests right now with simple ReachOctree env to see...

If it is indeed the case, I would still stick to the current implementation and disable BatchNorm by default.. mainly because stacking octrees/point clouds in 3D volume would be too costly. Furthermore, it would require to stack 8^n frames if keeping 1:1 aspect ratio is of importance.

**Comment by AndrejOrsula** [comment]
image

Stacking of frames does not seem to have negative influence on batch normalization (gray - stacked with bn, red - not stacked with bn)
